wdpk/entware/cgi-bin/entware.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+import cgitb
+
+cgitb.enable()
+
+import cgi
+import os
+import re
+import subprocess
+import sys
+import urllib.parse
+
+os.environ["PATH"] = os.path.pathsep.join(["/opt/bin", os.environ["PATH"]])
+REQUEST_METHOD = os.environ["REQUEST_METHOD"]
+QUERY_STRING = os.environ["QUERY_STRING"]
+
+
+def is_post():
+    return REQUEST_METHOD == "POST"
+
+
+def get_package_name(qs):
+    package_name = qs.get("package_name", [])
+    package_name = package_name[0] if package_name else ""
+
+    if not re.match(r"^\w+$", package_name):
+        print(f"Invalid package name {package_name!r}")
+        exit(1)
+
+    return package_name
+
+
+def print_output(output):
+    for line in output.decode().splitlines():
+        print(f"<p>{line}</p>")
+
+
+def search(qs):
+    package_name = get_package_name(qs)
+    output = subprocess.check_output(["opkg", "find", f"{package_name}*"])
+    print_output(output)
+
+
+def install_remove(qs, method):
+    package_name = get_package_name(qs)
+    output = subprocess.check_output(["opkg", method, f"{package_name}"])
+    print_output(output)
+    print('Done')
+
+def install(qs):
+    install_remove(qs, 'install')
+
+
+def remove(qs):
+    install_remove(qs, 'remove')
+
+
+FUNCTIONS = {"search": search, "install": install, "remove": remove}
```

Comment:
would be nice to be able to list the installed packages too

Reply:
implemented here https://github.com/WDCommunity/wdpksrc/pull/87/commits/292588580adcff53baaf23dd6a474e217d310fd4